src/clients/crux-wallet-onboarding-client.ts

```diff
@@ -0,0 +1,104 @@
+    private _cruxDomainRepository: ICruxDomainRepository;
+    private _configKeyManager?: IKeyManager;
+    private _cruxDomain?: CruxDomain;
+    private _domainContext?: CruxDomainId;
+    constructor(options: ICruxOnBoardingClientOptions) {
+        setCacheStorage(options.cacheStorage || new InMemStorage());
+        // set the repository constructors to be used further in the client
+        this._cruxDomainRepository = new BlockstackCruxDomainRepository();
+        // set the keyManagers if available
+        this._configKeyManager =  typeof options.configKey === "string" ? new BasicKeyManager(options.configKey) : options.configKey;
+        // set the domainContext if domain is provided
+        this._domainContext = options.domain ? new CruxDomainId(options.domain) : undefined;
+        this._initPromise = this._init();
+        log.info("CruxOnBoardingClient initialised");
+    }
+    set domain(domain: string) {
+        this._domainContext = new CruxDomainId(domain);
+    }
+    public isCruxDomainAvailable = async (domain: string): Promise<boolean> => {
+        return this._cruxDomainRepository.find(new CruxDomainId(domain));
+    }
+    public registerCruxDomain = async (domain: string): Promise<void> => {
+        // TODO: implementation of auto registration of domain on blockchain
+        throw ErrorHelper.getPackageError(null, PackageErrorCode.IsNotSupported);
+    }
+    public getCruxDomainState = async (): Promise<DomainRegistrationStatus> => {
+        await this._initPromise;
+        return this._getCruxDomain().status;
+    }
+    public getNameServiceConfig = async (): Promise<IBlockstackServiceInputOptions|undefined> => {
+        await this._initPromise;
+        return this._getCruxDomain().config.nameserviceConfiguration;
+    }
+    public getAssetMapping = async (): Promise<IClientAssetMapping> => {
+        await this._initPromise;
+        return this._getCruxDomain().config.assetMapping;
+    }
+    public getAssetList = async (): Promise<IGlobalAssetList> => {
+        await this._initPromise;
+        return this._getCruxDomain().config.assetList;
+    }
+    public putNameServiceConfig = async (newNameServiceConfig: IBlockstackServiceInputOptions): Promise<void> => {
+        await this._initPromise;
+        const cruxDomain = this._getCruxDomain();
+        cruxDomain.config.nameserviceConfiguration = newNameServiceConfig;
+        this._cruxDomain = await this._cruxDomainRepository.save(cruxDomain, this._getConfigKeyManager());
+        return;
+    }
+    public putAssetMapping = async (newAssetMapping: IClientAssetMapping): Promise<void> => {
+        await this._initPromise;
+        const cruxDomain = this._getCruxDomain();
+        cruxDomain.config.assetMapping = newAssetMapping;
+        this._cruxDomain = await this._cruxDomainRepository.save(cruxDomain, this._getConfigKeyManager());
+        return;
+    }
+    public putAssetList = async (newAssetList: IGlobalAssetList): Promise<void> => {
```

Comment:
> putAssetList

We don't want to separately expose this method. Please populate assetList appropriately at the time of putAssetMapping.

Reply:
So basically, only when assetMap is updated, will assetList be updated.